File: app/src/main/java/com/bigfriedicecream/recipes/presenters/MainPresenter.java
package com.bigfriedicecream.recipes.presenters;


import com.bigfriedicecream.recipes.interfaces.IMainContract;
import com.bigfriedicecream.recipes.observables.EventsDispatcher;

import java.util.HashMap;
import java.util.Observable;
import java.util.Observer;

public class MainPresenter implements IMainContract.Presenter, Observer {

    private IMainContract.View view;
    private EventsDispatcher eventsRepository;

    public MainPresenter(IMainContract.View v) {
        view = v;
    }

    public void start() {
        eventsRepository = EventsDispatcher.getInstance();
        eventsRepository.addObserver(this);
        view.render();
    }

    public void stop() {
        eventsRepository.deleteObserver(this);
    }

    @Override
    public void update(Observable observable, Object o) {
        HashMap<String, String> hashMap = (HashMap<String, String>)o;
        if (observable instanceof EventsDispatcher) {
            System.out.println(hashMap.get("name") + hashMap.get("id"));
        }
    }
}

File: Code/app/src/main/java/com/twobrothers/overcooked/presenters/MainPresenter.kt
package com.twobrothers.overcooked.presenters

import com.twobrothers.overcooked.interfaces.IMainContract
import com.twobrothers.overcooked.observables.EventsDispatcher

import java.util.HashMap
import java.util.Observable
import java.util.Observer


class MainPresenter(private val view: IMainContract.View) : IMainContract.Presenter, Observer {
    private val eventsDispatcher:EventsDispatcher = EventsDispatcher.instance

    override fun start() {
        eventsDispatcher.addObserver(this)
    }

    override fun stop() {
        eventsDispatcher.deleteObserver(this)
    }

    override fun load() {
        view.renderList()
    }

    override fun update(observable: Observable, o: Any) {
        val hashMap:HashMap<String, String> = o as HashMap<String, String>
        if (observable is EventsDispatcher) {
            if (hashMap["name"] == "NAVIGATION_RECIPE") {
                val id:String = hashMap["id"]!!
                view.renderRecipe(id)
            }
        }
    }
}
